File: data/test_cases/sql/vulnerable_sql.py
# Vulnerable: violates SQL-001 / SQL-002 (unbound user input concatenated into SQL)
@app.get("/users/{user_id}")
async def get_user(user_id: int):
    query = f"SELECT * FROM users WHERE id = {user_id}"
    return await db.execute(query)

File: data/test_cases/sql/vulnerable_sql.js
// Vulnerable: violates SQL-001 / SQL-002 (string interpolation with user-controlled input)
const express = require('express');
const router = express.Router();

router.get('/users/:userId', async (req, res) => {
  const userId = req.params.userId;
  const query = `SELECT * FROM users WHERE id = ${userId}`;
  const result = await pool.query(query);
  res.json(result.rows);
});

module.exports = router;
